### DISCUSSION TITLE
Inconsistent Middleware Configuration Behavior with Different Export Syntax

### DISCUSSION THREAD

**Discussion by antoineducrot** [vercel/next.js#67662] [Help]
### Description

Hi,

I'm experiencing an issue with middleware configuration in my Next.js project. I've noticed that the behavior of my middleware differs depending on how I export the `config` object. Here are the details:

### Working Version

```typescript
import type { NextRequest } from 'next/server';
import { authMiddleware } from './middleware/auth.middleware';

const middleware = (request: NextRequest) => {
  console.log(request.url);
  return authMiddleware(request);
};

export const config = {
  matcher: ['/((?!api|_next/static|_next/image|.*\\..*).*)'],
};

export { middleware };
```

### Non-Working Version

```typescript
import type { NextRequest } from 'next/server';
import { authMiddleware } from './middleware/auth.middleware';

const middleware = (request: NextRequest) => {
  console.log(request.url);
  return authMiddleware(request);
};

const config = {
  matcher: ['/((?!api|_next/static|_next/image|.*\\..*).*)'],
};

export { config, middleware };
```

### Issue

In the first version, where config is exported directly with export const config, the middleware works as expected. However, in the second version, where config is defined first and then exported with export { config, middleware }, the middleware does not function correctly.

### Expected Behavior

I expect both versions to work similarly, as they both export the same config and middleware objects.

### Steps to Reproduce

Set up a Next.js project with middleware.

Use the first version of the code above; observe that it works.

Switch to the second version; observe that it does not work.

### Question

Is this behavior intentional or a bug? Should the export style affect the functionality of the middleware configuration in Next.js?

Thank you for your assistance!

**Comment by icyJoseph** [comment]
Hi,

I think you can open a bug with this. The entire middleware file is parsed into an AST, and then the config is extracted by "querying" the AST. 

IIRC, just came out of vacation, so I might be off, but a function like this, https://github.com/vercel/next.js/blob/96092dd42dc8d7fe4ef0b6ea9889020b2da2af86/packages/next/src/build/analysis/extract-const-value.ts is used to extract `export const config`, and from there get the `matcher` for middleware. I can search a bit more in a while.
